src/operations.jl

```diff
@@ -44,6 +44,22 @@ function *(op::AbstractLinearOperator{T}, v::AbstractVector{S}) where {T, S}
   return res
 end
 
+function *(
+  v::Union{Adjoint{S, V}, Transpose{S, V}},
+  op::AbstractLinearOperator{T},
+) where {T, S, V <: AbstractVector{S}}
+  nrow, ncol = size(op)
+  res = similar(v.parent, promote_type(T, S), ncol)
+  tv = typeof(v)
+  if tv <: Adjoint
+    mul!(res, adjoint(op), adjoint(v))
+    return adjoint(res)
+  else
+    mul!(res, transpose(op), transpose(v))
+    return transpose(res)
+  end 
+end
```

Comment:
I think it's more julian to use the multiple dispatch trick here. So, you can split this function in 2, the first one would be:
```
function *(
  v::Adjoint{S, V},
  op::AbstractLinearOperator{T},
) where {T, S, V <: AbstractVector{S}}
  nrow, ncol = size(op)
  res = similar(v.parent, promote_type(T, S), ncol)
  mul!(res, adjoint(op), adjoint(v))
  return adjoint(res)
end
```
and the second one similar for `Transpose{S, V}`.

Reply:
I agree.